lib/llm/src/audit/sink/sink_nats.rs

```diff
@@ -0,0 +1,244 @@
+                                    if consecutive_failures >= self.cb_threshold {
+                                        error!(fails = consecutive_failures, "nats: consecutive failures; opening circuit breaker");
+                                        breaker_open_until = Some(Instant::now() + Self::jittered(self.cb_sleep));
+                                        // drop buffered messages to avoid unbounded growth
+                                        buf.clear();
+                                    }
+                                } else {
+                                    consecutive_failures = 0;
+                                }
+                            }
+                        }
+                        None => {
+                            // Channel closed: flush best-effort and exit task
+                            if let Err(e) = Self::flush(&self.js, &self.subject, &mut buf).await {
+                                warn!(error=%e, "nats: final flush failed on shutdown");
+                            }
+                            break;
+                        }
+                    }
+                }
+                _ = ticker.tick() => {
+                    if buf.is_empty() { continue; }
+                    if let Err(e) = Self::flush(&self.js, &self.subject, &mut buf).await {
+                        consecutive_failures += 1;
+                        warn!(error=%e, fails=consecutive_failures, "nats: flush(interval) failed");
+                        let exp = 2u32.saturating_pow(consecutive_failures.min(15));
+                        let mut backoff = self.backoff_base.saturating_mul(exp);
+                        if backoff > self.max_backoff { backoff = self.max_backoff; }
+                        time::sleep(Self::jittered(backoff)).await;
+                        if consecutive_failures >= self.cb_threshold {
+                            error!(fails = consecutive_failures, "nats: consecutive failures; opening circuit breaker");
+                            breaker_open_until = Some(Instant::now() + Self::jittered(self.cb_sleep));
+                            buf.clear();
+                        }
+                    } else {
+                        consecutive_failures = 0;
+                    }
+                }
+            }
+        }
+    }
+}
+
+/// Public sink type – thin wrapper around a bounded queue to a background worker.
+pub struct NatsSink {
+    tx: mpsc::Sender<Vec<u8>>,
+}
+
+impl NatsSink {
+    /// Create a new NatsSink using the shared NATS client from DistributedRuntime.
+    /// Returns None if no NATS client is available.
+    pub fn new(nats_client: Option<&dynamo_runtime::transports::nats::Client>) -> Option<Self> {
+        let Some(client) = nats_client else {
+            warn!(
+                "NATS sink requested but no DistributedRuntime NATS client available; skipping NATS audit sink"
+            );
+            return None;
```

Comment:
Instead of taking `Option` and returning `Option`, check before calling `new` and pass/return the values directly.